### ISSUE TITLE
Bug - modulation range does not reset when changing a cable's destination

### ISSUE THREAD

**Issue by PowerUser64** [BespokeSynth/BespokeSynth#1139]
## Describe the bug

When changing a modulation cable's destination, the modulation range of the modulator doesn't reset in the same way it would if the cable was being initially dragged.

## Build info

b6e01b9e3d13654c6849f1389767b459e7c5d6a3

## Steps To Reproduce

[bespoke modulation range bug.webm](https://user-images.githubusercontent.com/24578572/216567865-e2981c5a-3736-4d7e-8184-b653125a7bc4.webm)

(notice how the values on the macroslider's range don't change when the cable's destination is changed)

## Expected vs. actual behavior

The modulation range should get reset when the cable is connected to a new destination as if it was freshly connected. I don't have an example, but I have noticed this can be fairly problematic sometimes.

**Comment by merkys** [comment]
This would indeed be nice to have. Fix is localized in `IModulator::OnModulatorRepatch()` and is not that difficult to implement. However, modulators may have multiple targets. What should be done if their ranges differ? I guess we should take the range supported by all the targets, thus the shortest one. Or is there a better option?

**Comment by PowerUser64** [comment]
I think having the range follow the first-connected destination is probably the best way to go. Picking the smallest range seems smart, but I think it could probably lead to a more confusing user experience if there's a cable going to a lot of destinations. (And, it would be impossible to change the upper bound sometimes, like if there's a check box as a destination.)

At some point, there should probably be some kind of a "range remap" feature that allows remapping the range of each modulation cable on a per-destination basis, like what Vital or Serum have in their mod matrix. This way, the range of the modulator is managed at the destination rather than the source, and options like whether to clamp the range could be exposed. Maybe it could take over the "right-click to add an LFO" feature when a cable is connected to something and spawn something like a modified `curve` module instead. This would be a more complete way to solve this problem (IMO), although it'd also be a lot more work to implement.

**Comment by merkys** [comment]
I agree that multi-target modulation is a topic needing more thought/discussion or even larger interface changes. I like the idea of modulation value management at the target, although this might make some UI features less intuitive.

Let me focus on the current bug report. Currently replacing one target with another one does not affect modulator's modulation range. This can be circumvented by deleting the cable in between switching targets, but such behavior is clunky. I think I know how to fix it.

**Comment by PowerUser64** [comment]
It may also be worth testing out a few different methods for the range picker and seeing what works best. The "first come, first served" method that I suggested, the minimum method, like you suggested, and maybe even a maximum method, would all be good ones to test. Right now, splitting modulation cables is a bit janky, but I could see it being more useful if there's a good range picking algorithm.

**Comment by merkys** [comment]
It makes sense to try out different approaches and see which of them is more intuitive. For now I have just opened PR #1805 to adjust the range after the change of single-targetted modulator.
